pkg/reconciler/openshift/tektonresult/extension.go

```diff
@@ -76,6 +84,7 @@ func (oe openshiftExtension) Transformers(comp v1alpha1.TektonComponent) []mf.Tr
 		occommon.RemoveRunAsUser(),
 		occommon.RemoveRunAsGroup(),
 		occommon.ApplyCABundles,
+		injectBoundSAToken(comp),
```

Comment:
i think we should get the value of field required for this before and only pass the field required in transformer function

Reply:
or add this function based on the condition

Reply:
We already have a condition in injectBoundSAToken and we would need it anyway to apply only to a particular deployment. I feel having a condition introduces a bad coupling between functions.

Reply:
So I am passing properties now.